### PR TITLE
[SPARK-38949][SQL] Wrap SQL statements by double quotes in error messages

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to wrap any SQL statement in error messages by double quotes "", and apply new implementation of `QueryErrorsBase.toSQLStmt()` to all exceptions in `Query.*Errors` w/ error classes. Also this PR modifies all affected tests, see the list in the section "How was this patch tested?".

### Why are the changes needed?
To improve user experience with Spark SQL by highlighting SQL statements in error massage and make them more visible to users.

### Does this PR introduce _any_ user-facing change?
Yes. The changes might influence on error messages that are visible to users.

Before:
```sql
The operation DESC PARTITION is not allowed
```

After:
```sql
The operation "DESC PARTITION" is not allowed
```

### How was this patch tested?
By running affected test suites:
```
$ build/sbt "sql/testOnly *QueryExecutionErrorsSuite"
$ build/sbt "sql/testOnly *QueryParsingErrorsSuite"
$ build/sbt "sql/testOnly *QueryCompilationErrorsSuite"
$ build/sbt "test:testOnly *QueryCompilationErrorsDSv2Suite"
$ build/sbt "test:testOnly *ExtractPythonUDFFromJoinConditionSuite"
$ build/sbt "testOnly *PlanParserSuite"
$ build/sbt "sql/testOnly *SQLQueryTestSuite -- -z transform.sql"
$ build/sbt "sql/testOnly *SQLQueryTestSuite -- -z join-lateral.sql"
$ build/sbt "sql/testOnly *SQLQueryTestSuite -- -z describe.sql"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@ivoson @yutoacts @leesf @cloud-fan @HyukjinKwon @gengliangwang @panbingkun Could you review this PR, please.

**Review by HyukjinKwon — APPROVED** [review]
LGTM from my side.

**Review by leesf — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala [hunk 1]
```diff
@@ -416,13 +437,20 @@ object QueryParsingErrors extends QueryErrorsBase {
...
+      ctx)
   }
 
   def defineTempFuncWithIfNotExistsError(ctx: CreateFunctionContext): Throwable = {
-    new ParseException("INVALID_SQL_SYNTAX",
-      Array("It is not allowed to define a TEMPORARY function with IF NOT EXISTS."), ctx)
+    new ParseException(
+      errorClass = "INVALID_SQL_SYNTAX",
+      messageParameters = Array(
+        s"It is not allowed to define a ${toSQLStmt("TEMPORARY")} function" +
```
Comment (ivoson) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:452]: I am wondering that should the `temporary function` here be the sql statement?

**Review by ivoson — APPROVED** [review]

**Review by yutoacts — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala (hunk 1 above)
Comment (MaxGekk) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:452]: makes sense. Let's do that.

**Review by cloud-fan — APPROVED** [review]

**Comment by MaxGekk** [comment]
GAs passed. Merging to master.
Thank you, @ivoson @yutoacts @leesf @cloud-fan @HyukjinKwon @panbingkun for review.

**Pull request closed without merging**